napari/components/_tests/test_dims.py

```diff
@@ -339,6 +339,15 @@ def test_changing_focus():
     assert dims.last_used == 2
 
 
+def test_changing_focus_changing_ndisplay():
+    dims = Dims(ndim=4, ndisplay=2)
+    # simulates putting focus from slider 0 to slider 1
+    dims.last_used = 1
```

Comment:
As the `last_used` setter does some validation now, it would be good to assert that setting the valid value works as expected.

```suggestion
    dims.last_used = 1
    assert dims.last_used == 1
```

That also sets an informative precondition before taking the action of changing `ndisplay`.